streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/worker/context/RayRuntimeContext.java

```diff
@@ -16,6 +27,15 @@
   private Long batchId;
   private final Long maxBatch;
   private Map<String, String> config;
+  /**
+   * Backend for keyed state. This might be empty if we're not on a keyed stream.
+   */
+  protected transient KeyStateBackend keyStateManager;
```

Comment:
Actually, from within the design document I found quite a few terminology inconsistent. Such as this "Backend" vs "Manager" and "BatchID" vs "CheckpointID". I believe a consistent terminology is the least requirement for any further discussion or we're in somewhere continuously confusing.

The architecture *seems like*

A set of `State`s  share one `CheckpointID`
    ↓  handled by 
`StateStrategy`(nope, I think it should be at least named as StateStoreStrategy)
    ↓  stored into
`StateBackend`

so expect of these four terminologies(State, CheckpointID, StateStrategy, StateBackend), other phrases like `IStore` are quite confusing. What exactly it store?

Besides, if this pull request focuses on State, the Checkpoint is an external logic to organize the state. We should properly mention it in the PR description instead of let the reviewers **guess** what this  PR wants to do.

Reply:
yes, the terminologies will be updated. I'll check the other codes. Basically, the state is a way to reach exactly once streaming computing, therefor the checkpoint is bounded to the state management.